client/ayon_nuke/api/plugin.py

```diff
@@ -304,55 +346,91 @@ def integrate_links(self, node, outputs=True):
         for dep_nodes in dependent_nodes:
             dep_nodes.setInput(0, node)
 
-    def set_selected_nodes(self, pre_create_data):
-        if pre_create_data.get("use_selection"):
-            selected_nodes = nuke.selectedNodes()
-            if selected_nodes == []:
-                raise NukeCreatorError("Creator error: No active selection")
-            elif len(selected_nodes) > 1:
-                NukeCreatorError("Creator error: Select only one camera node")
-            self.selected_node = selected_nodes[0]
-        else:
-            self.selected_node = None
+    def _get_current_selected_nodes(
+            self,
+            pre_create_data,
+            class_name: str = None,
+        ):
```

Comment:
```suggestion
        self,
        pre_create_data,
        class_name: str = None,
    ):
```

Reply:
Yeah here as well, that's how things were implemented initially so I didn't really question it to be honest. 
I'll move the check out of this function to keep things consistent.